From e379343e404aa3f100d63c577922435039a388dd Mon Sep 17 00:00:00 2001
From: Lothar Rubusch <lothar.rubusch@enclustra.com>
Date: Fri, 24 Nov 2023 08:47:08 +0000
Subject: [PATCH 2/8] Add Enclustra devicetree to Makefile

Signed-off-by: Andreas Buerkler <andreas.buerkler@enclustra.com>
Signed-off-by: Lothar Rubusch <lothar.rubusch@enclustra.com>
---
 arch/arm/dts/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 62d48b4a45..83f579713c 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -457,7 +457,8 @@ dtb-$(CONFIG_ARCH_SOCFPGA) +=				\
 	socfpga_n5x_socdk.dtb				\
 	socfpga_stratix10_socdk.dtb			\
 	socfpga_stratix10_socdk_nand.dtb                \
-	socfpga_stratix10_socdk_qspi.dtb
+	socfpga_stratix10_socdk_qspi.dtb		\
+	enclustra-user.dtb				\
 
 dtb-$(CONFIG_TARGET_DRA7XX_EVM) += dra72-evm.dtb dra7-evm.dtb	\
 	dra72-evm-revc.dtb dra71-evm.dtb dra76-evm.dtb
-- 
2.34.1

