diff -purN a/board/enclustra/bootscripts/qspi-aa1.cmd b/board/enclustra/bootscripts/qspi-aa1.cmd
--- a/board/enclustra/bootscripts/qspi-aa1.cmd	2023-12-15 13:39:41.342530084 +0000
+++ b/board/enclustra/bootscripts/qspi-aa1.cmd	2023-12-15 15:26:49.923217129 +0000
@@ -3,4 +3,5 @@ sf probe
 sf read ${ram_addr_kernel} ${qspi_offset_addr_kernel} ${size_kernel}
 sf read ${ram_addr_devicetree} ${qspi_offset_addr_devicetree} ${size_devicetree}
 sf read ${ram_addr_rootfs} ${qspi_offset_addr_rootfs} ${size_rootfs}
-bootm ${ram_addr_kernel} ${ram_addr_rootfs} ${ram_addr_devicetree}
+run ${fdtload}
+bootm ${ram_addr_kernel} ${ram_addr_rootfs} ${ram_addr_devicetree} - ${ram_addr_devicetree}
diff -purN a/board/enclustra/bootscripts/qspi.cmd b/board/enclustra/bootscripts/qspi.cmd
--- a/board/enclustra/bootscripts/qspi.cmd	2023-12-15 13:39:41.342530084 +0000
+++ b/board/enclustra/bootscripts/qspi.cmd	2023-12-15 15:36:41.539778503 +0000
@@ -5,4 +5,5 @@ bridge enable
 sf read ${ram_addr_kernel} ${qspi_offset_addr_kernel} ${size_kernel}
 sf read ${ram_addr_devicetree} ${qspi_offset_addr_devicetree} ${size_devicetree}
 sf read ${ram_addr_rootfs} ${qspi_offset_addr_rootfs} ${size_rootfs}
-bootm ${ram_addr_kernel} ${ram_addr_rootfs} ${ram_addr_devicetree}
+run ${fdtload}
+bootm ${ram_addr_kernel} ${ram_addr_rootfs} ${ram_addr_devicetree} - ${ram_addr_devicetree}
diff -purN a/board/enclustra/bootscripts/sd-aa1.cmd b/board/enclustra/bootscripts/sd-aa1.cmd
--- a/board/enclustra/bootscripts/sd-aa1.cmd	2023-12-15 13:39:41.342530084 +0000
+++ b/board/enclustra/bootscripts/sd-aa1.cmd	2023-12-15 15:37:26.231218580 +0000
@@ -5,4 +5,5 @@ fatload mmc 0:1 ${ram_addr_rootfs} uramd
 
 altera_set_storage QSPI
 
-bootm ${ram_addr_kernel} ${ram_addr_rootfs} ${ram_addr_devicetree}
+run ${fdtload}
+bootm ${ram_addr_kernel} ${ram_addr_rootfs} ${ram_addr_devicetree} - ${ram_addr_devicetree}
diff -purN a/board/enclustra/bootscripts/sd.cmd b/board/enclustra/bootscripts/sd.cmd
--- a/board/enclustra/bootscripts/sd.cmd	2023-12-15 13:39:41.342530084 +0000
+++ b/board/enclustra/bootscripts/sd.cmd	2023-12-15 15:40:32.232892808 +0000
@@ -3,4 +3,5 @@ fpga load 0 ${ram_addr_bitstream} 0x${fi
 bridge enable
 fatload mmc 0:1 ${ram_addr_kernel} uImage
 fatload mmc 0:1 ${ram_addr_devicetree} devicetree.dtb
+run ${fdtload}
 bootm ${ram_addr_kernel} - ${ram_addr_devicetree}
diff -purN a/include/configs/socfpga_mercury_aa1.h b/include/configs/socfpga_mercury_aa1.h
--- a/include/configs/socfpga_mercury_aa1.h	2023-12-15 13:39:41.610526801 +0000
+++ b/include/configs/socfpga_mercury_aa1.h	2023-12-15 16:02:33.360466279 +0000
@@ -44,7 +44,10 @@
     "ram_addr_rootfs=0x12000000\0" \
     "ram_addr_kernel=0x11000000\0" \
     "ram_addr_devicetree=0x10000000\0" \
+    "ram_addr_devicetree_overlay=0x100c0000\0" \
     "ram_addr_boot-script=0x10800000\0" \
+    "fdtofile=socfpga_enclustra_mercury_XXX_overlay.dtbo\0" \
+    "fdtload=load ${devtype} ${distro_bootpart} ${ram_addr_devicetree} ${bootdir}/${fdtfile}; load ${devtype} ${distro_bootpart} ${ram_addr_devicetree_overlay} ${bootdir}/${fdtofile}; fdt addr $ram_addr_devicetree; fdt resize 8191; fdt apply $ram_addr_devicetree_overlay\0" \
     "bootm_size=0x0a000000\0" \
     "bootargs-qspi=earlycon console=ttyS0,115200 rw root=/dev/ram0\0" \
     "bootargs=earlycon console=ttyS0,115200 rw root=/dev/mmcblk0p3\0" \
diff -purN a/include/configs/socfpga_mercury_sa1.h b/include/configs/socfpga_mercury_sa1.h
--- a/include/configs/socfpga_mercury_sa1.h	2023-12-15 13:39:41.610526801 +0000
+++ b/include/configs/socfpga_mercury_sa1.h	2023-12-15 15:52:26.064006503 +0000
@@ -34,7 +34,10 @@
     "ram_addr_rootfs=0x12000000\0" \
     "ram_addr_kernel=0x11000000\0" \
     "ram_addr_devicetree=0x10000000\0" \
+    "ram_addr_devicetree_overlay=0x100c0000\0" \
     "ram_addr_boot-script=0x10800000\0" \
+    "fdtofile=socfpga_enclustra_mercury_XXX_overlay.dtbo\0" \
+    "fdtload=load ${devtype} ${distro_bootpart} ${ram_addr_devicetree} ${bootdir}/${fdtfile}; load ${devtype} ${distro_bootpart} ${ram_addr_devicetree_overlay} ${bootdir}/${fdtofile}; fdt addr $ram_addr_devicetree; fdt resize 8191; fdt apply $ram_addr_devicetree_overlay\0" \
     "bootm_size=0x0a000000\0" \
     "bootargs-qspi=earlycon console=ttyS0,115200 rw root=/dev/ram0\0" \
     "bootargs=earlycon console=ttyS0,115200 rw rootwait root=/dev/mmcblk0p3\0" \
diff -purN a/include/configs/socfpga_mercury_sa2.h b/include/configs/socfpga_mercury_sa2.h
--- a/include/configs/socfpga_mercury_sa2.h	2023-12-15 13:39:41.610526801 +0000
+++ b/include/configs/socfpga_mercury_sa2.h	2023-12-15 16:02:31.576488417 +0000
@@ -34,7 +34,10 @@
     "ram_addr_rootfs=0x12000000\0" \
     "ram_addr_kernel=0x11000000\0" \
     "ram_addr_devicetree=0x10000000\0" \
+    "ram_addr_devicetree_overlay=0x100c0000\0" \
     "ram_addr_boot-script=0x10800000\0" \
+    "fdtofile=socfpga_enclustra_mercury_XXX_overlay.dtbo\0" \
+    "fdtload=load ${devtype} ${distro_bootpart} ${ram_addr_devicetree} ${bootdir}/${fdtfile}; load ${devtype} ${distro_bootpart} ${ram_addr_devicetree_overlay} ${bootdir}/${fdtofile}; fdt addr $ram_addr_devicetree; fdt resize 8191; fdt apply $ram_addr_devicetree_overlay\0" \
     "bootm_size=0x0a000000\0" \
     "bootargs-qspi=earlycon console=ttyS0,115200 rw root=/dev/ram0\0" \
     "bootargs=earlycon console=ttyS0,115200 rw rootwait root=/dev/mmcblk0p3\0" \
