From 42302d0c760c14e6e3518f2c194a3e20ba919731 Mon Sep 17 00:00:00 2001
From: Lothar Rubusch <lothar.rubusch@enclustra.com>
Date: Wed, 31 Jan 2024 15:47:02 +0000
Subject: [PATCH] board: add rootwait for AA1

Signed-off-by: Lothar Rubusch <lothar.rubusch@enclustra.com>
---
 include/configs/socfpga_mercury_aa1.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/socfpga_mercury_aa1.h b/include/configs/socfpga_mercury_aa1.h
index 997d0929e2..dbf4ae04e1 100644
--- a/include/configs/socfpga_mercury_aa1.h
+++ b/include/configs/socfpga_mercury_aa1.h
@@ -53,7 +53,7 @@
     "fdtload=load ${devtype} ${distro_bootpart} ${ram_addr_devicetree} ${bootdir}/${fdtfile}; load ${devtype} ${distro_bootpart} ${ram_addr_devicetree_overlay} ${bootdir}/${fdtofile}; fdt addr $ram_addr_devicetree; fdt resize 8191; fdt apply $ram_addr_devicetree_overlay\0" \
     "bootm_size=0x0a000000\0" \
     "bootargs-qspi=earlycon console=ttyS0,115200 rw root=/dev/ram0\0" \
-    "bootargs=earlycon console=ttyS0,115200 rw root=/dev/mmcblk0p2\0" \
+    "bootargs=earlycon console=ttyS0,115200 rw root=/dev/mmcblk0p2 rootwait\0" \
     "qspiload=setenv bootargs ${bootargs-qspi}; sf probe; sf read ${ram_addr_boot-script} ${qspi_offset_addr_boot-script} ${size_boot-script}\0" \
     "qspiboot=source ${ram_addr_boot-script}\0"
 
-- 
2.34.1

