From a98964cbbeddfce82b2dffdf060c34446b78108b Mon Sep 17 00:00:00 2001
From: Lothar Rubusch <lothar.rubusch@enclustra.com>
Date: Tue, 13 Feb 2024 15:19:09 +0000
Subject: [PATCH] dt: fix dtbo for qspiboot on sa1/2

Signed-off-by: Lothar Rubusch <lothar.rubusch@enclustra.com>
---
 board/enclustra/bootscripts/qspi.cmd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/board/enclustra/bootscripts/qspi.cmd b/board/enclustra/bootscripts/qspi.cmd
index 0c715b5ddc..393c7a30f7 100644
--- a/board/enclustra/bootscripts/qspi.cmd
+++ b/board/enclustra/bootscripts/qspi.cmd
@@ -3,6 +3,6 @@ sf read ${ram_addr_bitstream} ${qspi_offset_addr_bitstream} ${size_bitstream}
 fpga load 0 ${ram_addr_bitstream} ${size_bitstream}
 bridge enable
 sf read ${ram_addr_kernel} ${qspi_offset_addr_kernel} ${size_kernel}
-sf read ${ram_addr_devicetree} ${qspi_offset_addr_devicetree} ${size_devicetree}
 sf read ${ram_addr_rootfs} ${qspi_offset_addr_rootfs} ${size_rootfs}
+run fdtload_qspi
 bootm ${ram_addr_kernel} ${ram_addr_rootfs} ${ram_addr_devicetree}
-- 
2.34.1

